### ISSUE TITLE
y_bitmap text enhancement

### ISSUE THREAD

**Issue by Crayder** [pawn-lang/YSI-Includes#88]
Currently the text rendering is pretty static. The only thing we can customize is the color of the text.

I seen how it's layed out, with each letter's data and stuff. But is it possible to read this data from a font or something? That would make this a lot more flexible.

I just want to be able to resize, color, and bold the text. Obviously it's not just that simple but it has been done before.

**Comment by Southclaws** [comment]
I think this is far beyond the scope of Pawn to be honest, stuff like this can be left to libraries that have already done all the hard work like @python-pillow or Cairo.

**Comment by Crayder** [comment]
Yeah I was looking for an easy way to do it, and there isn't much.

**Comment by Y-Less** [comment]
I would really agree with Southclaw.  I wrote y_bitmap to dump basic data out from the server visually, not for a fully fledged graphics processing library (it doesn't even have loading - that would be the only other feature I would think would be important, imagine drawing maps and loading them for gang zones).  Anyway, even if you did decide to implement this, I would suspect that it would be a totally different storage library dealing with font files, and using the basic primatives of y_bitmap to draw the data on, rather than a modification to this library.

**Comment by Southclaws** [comment]
@Crayder if you're actively looking for a graphics solution you might be interested in my Pawn/Python plugin I recently made, it's not finished yet but ideally you could write your image generation script in Python using Pillow and call it with your parameters from Pawn.

**Comment by Crayder** [comment]
@Y-Less Yeah, if I did at all it would be a separate library specifically for fonts. That just seems quite far fetched for SA-MP. I just tend to use y_bitmap a lot and some more font options would be nice.

@Southclaw I didn't notice that plugin, sounds interesting. I'll check it out.
Edit: Checked it out, looks great! How is the speed? I'll test it out real quick (well, I can't right now, don't have VS on my laptop and not home).

**Comment by Y-Less** [comment]
The simple way would be just draw some more bitmapped fonts.  I saw a brilliant way to do this recently, give me a second...

**Comment by Y-Less** [comment]
``` pawn
new g_myFont[256][8] = {
    {
        ________
        ___X____
        __X_X___
        __X_X___
        _XXXXX__
        _X___X__
        X_____X_
        ________
    },
    {
        ________
        XXXXX___
        X____X__
        XXXXX___
        X____X__
        X____X__
        XXXXX___
        ________
    },
    {
        ________
        __XXX___
        _X___X__
        X_______
        X_______
        _X___X__
        __XXX___
        ________
    },
    ...
};
```

Using this file:

https://gist.github.com/Y-Less/f64cdb1925c286bc9bf2ef6cbb8d191a

Which makes them already fairly clear in the defintions, and even clearer if you use a document map:

![fonts](https://cloud.githubusercontent.com/assets/991078/14649908/167bff6a-0661-11e6-87f2-59d23b5e9a81.png)

**Comment by Crayder** [comment]
That's pretty interesting. However, defining font sizes and all of the definitions needed for the different font sizes would be extensive. It is still clever though.

**Comment by Y-Less** [comment]
I would just generate them.
